src/__tests__/index.test.ts

```
@@ -0,0 +1,557 @@
+      createResource();
+      expect(resources).toHaveLength(2);
+      
+      cleanupResources();
+      expect(resources).toHaveLength(0);
+    });
+
+    it('should handle memory leaks prevention', () => {
+      let eventListeners = [];
+      
+      const addEventListener = (event: string, callback: Function) => {
+        eventListeners.push({ event, callback });
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)